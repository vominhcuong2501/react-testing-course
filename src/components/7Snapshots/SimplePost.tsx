export function SimplePost(props: {
    content: string,
    user: string,
    likesBy?: string[]
}) {
    function renderLikes() {
        if (props.likesBy) {
            return <div data-testid="likes-container">
                <h3>Likes by:</h3>
                <ul>
                    {
                        props.likesBy.map(like => {
                            return <li key={like}>{like}</li>
                        })
                    }
                </ul>
            </div>
        }
    }

    return <div data-testid="post-container">
        <h2>{props.user}:</h2>
        <p>{props.content}</p>
        {renderLikes()}
    </div>

}