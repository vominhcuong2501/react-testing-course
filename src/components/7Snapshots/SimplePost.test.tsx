import { SimplePost } from './SimplePost'
import { render } from '@testing-library/react'

describe('Simple post snapshot tests', ()=>{
    it('initial test', ()=>{
        const rendered = render(
            <SimplePost 
                content='Simple content'
                user='Alex'                
            />
        )
        expect(rendered.asFragment()).toMatchSnapshot()
    })
})