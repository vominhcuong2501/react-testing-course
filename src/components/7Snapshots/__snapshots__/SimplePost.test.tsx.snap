// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple post snapshot tests > initial test 1`] = `
<DocumentFragment>
  <div
    data-testid="post-container"
  >
    <h2>
      Alex:
    </h2>
    <p>
      Simple content
    </p>
  </div>
</DocumentFragment>
`;
